robyn/router.py

```diff
@@ -69,49 +75,76 @@ def _format_response(self, res):
                 body=str(res).encode("utf-8"),
             )
         return response
-
+    #certain methods like add_route and other decorators
+    #return self._add_route(HttpMethod.GET, endpoint, handler, const)
     def add_route(
         self,
         route_type: HttpMethod,
         endpoint: str,
-        handler: Callable,
+        handler: Callable, #hello world function
         is_const: bool,
+        dependencies: Dict[str,any],
         exception_handler: Optional[Callable],
     ) -> Union[Callable, CoroutineType]:
         @wraps(handler)
         async def async_inner_handler(*args):
+            signatureObj = (inspect.signature(handler))
+            argsFromHandler = signatureObj.parameters.values() #holds all args from func args
+            specificDep = dependencies.get(endpoint, dependencies["all"])
+            depToPass = [] 
+            paramList = [a.name for a in argsFromHandler]
+            if any(key not in specificDep for key in paramList):
+                missing_deps = [key for key in paramList if key not in specificDep]
+                raise ValueError("Parameter not specified in dependencies dictionary:", missing_deps)
```

Comment:
The error should be more like `Unknown Argument` as it is not necessary that the person wanted to refer to the dependency

Reply:
Will change it to `raise ValueError("Unknown Argument")`